src/settings/ui/frontmatterOptions.ts

```diff
@@ -73,5 +76,74 @@ export default class FrontmatterOptions extends IconFolderSetting {
           new Notice('...saved successfully');
         });
       });
+
+    new Setting(this.containerEl)
+      .setName('Refresh icons from frontmatter')
+      .setDesc(
+        'Sets the icon and color for each note in the vault based on the frontmatter properties. WARNING: This will change any manually set icons to the one defined in the frontmatter. IF A NOTE HAS NO FRONTMATTER, THE CURRENT ICON WILL BE REMOVED. Please restart Obsidian after this completes to see the changes.',
+      )
+      .addButton((btn) => {
+        btn.setButtonText('Refresh').onClick(async () => {
+          if (!this.plugin.getSettings().iconInFrontmatterEnabled) {
+            new Notice(
+              `[${config.PLUGIN_NAME}] Please enable "Use icon in frontmatter".`,
+            );
+            return;
+          }
+
+          new Notice(
+            `[${config.PLUGIN_NAME}] Refreshing icons from frontmatter, please wait...`,
+          );
+
+          const files = this.plugin.app.vault.getMarkdownFiles();
+
+          for (const file of files) {
+            const fileCache = this.plugin.app.metadataCache.getFileCache(file);
+
+            const frontmatterIconKey =
+              this.plugin.getSettings().iconInFrontmatterFieldName;
+            const frontmatterIconColorKey =
+              this.plugin.getSettings().iconColorInFrontmatterFieldName;
+
+            const iconName = fileCache.frontmatter?.[frontmatterIconKey];
+            let iconColor = fileCache.frontmatter?.[frontmatterIconColorKey];
+
+            if (!iconName) {
+              await this.plugin.removeFolderIcon(file.path);
+              continue;
+            }
+
+            if (typeof iconName !== 'string') {
+              const message = `[${config.PLUGIN_NAME}]\n${file.path}\nFrontmatter property type \`${frontmatterIconKey}\` has to be of type \`text\`.`;
+              logger.warn(message);
+              new Notice(message);
```

Comment:
The logger automatically places the plugin name in front of the name.

So, you can just remove the prefix in the original message and prepend it in the Notice.

Reply:
Should be good now. Thanks for making this plugin!